src/cfi.rs

```diff
@@ -766,7 +766,7 @@ impl Default for Augmentation {
 }
 
 impl Augmentation {
-    fn parse<'aug, 'bases, 'input, Endian, Section>(augmentation_str: &'aug str,
+    fn parse<'aug, 'bases, 'input, Endian, Section>(augmentation_str: &'aug [u8],
```

Comment:
So that we can trivially convert this to use the `Reader` trait. Since all of the characters that we understand currently are single-byte, it doesn't matter if we treat it as `&[u8]`. If we do need to handle multi-byte UTF-8 characters in future, we'll need to either potentially read the whole string into a `Vec` and then validate it as a UTF-8 string, or add a `read_utf8_char` method to the trait.

One thing we do lack from this approach is that we no longer validate that unknown augmentation strings are UTF-8. I don't see that as much of a problem though, since the user will handle this when they convert the `Reader` into a string.